src/classes/MageHero.tsx

```diff
@@ -19,10 +17,32 @@ export default class MageHero extends Hero {
     return this.mana;
   }
 
-  static fromJSON(json: MageHeroJSON): MageHero {
+  /*public static fromJSON(json: MageHeroJSON): MageHero {
     const abilities = json.abilities.map(a => new Ability(a.name, a.heroClassType, a.cooldown, a.cost, a.effect, () => {}));
     const potions = json.potions.map(p => new Potion(p.name, p.affectingField, p.affectingValue));
     const primaryWeapon = new WeaponItem(json.primaryWeapon.name, json.primaryWeapon.damage, json.primaryWeapon.heroClassType);
     return new MageHero(json.name, json.health, json.strength, json.armor, abilities, potions, primaryWeapon, json.mana, json.level);
-  } 
+  }*/
+
+  public static toJSON(playerName: string): MageHeroJSON {
```

Comment:
Can't you make this createCharacter or createCharacterJSON . toJSON should get the info from the `this`(the class itself)